packages/openapi-fetch/src/index.js

```diff
@@ -109,7 +109,7 @@ export default function createClient(clientOptions) {
 
     let id;
     let options;
-    let request = new CustomRequest(
+    let request = new Request(
```

Comment:
`CustomRequest` is also used in the middleware application lower down to perform an instanceof check. I think we should adjust that check as well for consistency.

The middleware specification in this case is arguably a bit broken, but I'd rather have that failing, than allowing for the inconsistent check. In the sense: At the moment, an onRequest middleware is required to return the same type of request than we send. If we do not adjust the check in this PR, it is required to return the type of the request configured on the client, even if a different request was provided. IMO that's very odd.